compiler/rustc_lint/src/levels.rs

```diff
@@ -1058,38 +1045,40 @@ impl<'s, P: LintLevelsProvider> LintLevelsBuilder<'s, P> {
     /// Returns `true` if the lint's feature is enabled.
     #[track_caller]
     fn check_gated_lint(&self, lint_id: LintId, span: Span, lint_from_cli: bool) -> bool {
-        if let Some(feature) = lint_id.lint.feature_gate {
-            if !self.features.active(feature) {
-                if self.lint_added_lints {
-                    let lint = builtin::UNKNOWN_LINTS;
-                    let (level, src) = self.lint_level(builtin::UNKNOWN_LINTS);
-                    // FIXME: make this translatable
-                    #[allow(rustc::diagnostic_outside_of_impl)]
-                    #[allow(rustc::untranslatable_diagnostic)]
-                    lint_level(
-                        self.sess,
+        // Exit early if the lint is ungated, or its feature is enabled.
+        let Some(feature) = lint_id.lint.feature_gate else { return true };
+        if self.features.active(feature) {
+            return true;
+        }
+
```

Comment:
I'm a little sad that we can't combine let-else with let-chains, so we need two separate early-exits here. Thankfully in this case the duplicated return is trivial.